Framework/Algorithms/inc/MantidAlgorithms/SortXAxis2.h

```diff
@@ -57,28 +52,27 @@ class DLLExport SortXAxis : public API::Algorithm {
 
   void sortIndicesByX(std::vector<std::size_t> &workspaceIndecies,
                       std::string order,
-                      Mantid::API::MatrixWorkspace_const_sptr inputWorkspace,
+                      const Mantid::API::MatrixWorkspace *inputWorkspace,
```

Comment:
It's generally better to prefer a reference here as it can't possibly be `null` so it doesn't need to be checked. Similar comment below...

Reply:
I had a performance decrease in testing when using a & over a * by almost a second on the unit test, it should never be null regardless right because you can't pass a null workspace through the algorithm factory or is that an incorrect assumption?

Reply:
I'd very very suprised if that a reference causes that much of a slow down on it's own. The idiom of passing by `const Type &` is extremely common in C++.  

You are correct that the factory should not return a null pointer but as a general point when writing functions it's generally considered good practice for input parameters to be `const &` and output parameters to be passed as pointers. This gives a distinction when calling them.

Reply:
I did some very interesting tests on this and at a compiler level the difference is at a very basic level 2 lines of assembly inside of the for loop (This was with an example not this code). So yeah you are right. I also re-ran the test and I am no longer getting the issue of longer test time, perhaps it was something I didn't change properly. Updated code accordingly.